jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/TaskConfiguration.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.gradle;
+
+import com.google.cloud.tools.jib.image.json.BuildableManifestTemplate;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.Task;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Nested;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.TaskAction;
+
+/**
+ * Defines the configuration parameters injected from {@link JibExtension}. {@link Task}s should
+ * extend this class.
+ */
+abstract class TaskConfiguration extends DefaultTask {
```

Comment:
Perhaps this can be a `@Nested` input type to the tasks that are extending it. `ImageConfig`? and you pass that to the task directly. Subjectively, I think avoiding this style of extending makes for better reading.

Reply:
Okay I'll just change it for Gradle then